client/packages/system/src/Report/api/hooks/useReportList.ts

```diff
@@ -35,24 +42,44 @@ export const useReportList = ({
   } = queryParams ?? {};
 
   const queryKey = [REPORT, storeId, LIST, sortBy, filterBy, offset];
-  const queryFn = async (): Promise<{
-    nodes: ReportRowFragment[];
-    totalCount: number;
-  }> => {
-    const query = await reportApi.reports({
-      filter: {
-        ...filterBy,
-        ...(context ? { context: { equalTo: context } } : null),
-        ...(subContext ? { subContext: { equalTo: subContext } } : null),
-      },
-      key: sortBy.key as ReportSortFieldInput,
-      desc: sortBy.isDesc,
-      storeId,
-    });
-    return {
-      nodes: query.reports.nodes,
-      totalCount: query.reports.totalCount,
-    };
+  const queryFn = async () => {
+    try {
+      const query = await reportApi.reports({
+        filter: {
+          ...filterBy,
+          ...(context ? { context: { equalTo: context } } : null),
+          ...(subContext ? { subContext: { equalTo: subContext } } : null),
+        },
+        key: sortBy.key as ReportSortFieldInput,
+        desc: sortBy.isDesc,
+        storeId,
+        userLanguage: language,
+      });
+
+      if (query?.reports?.__typename == 'ReportConnector') {
+        return {
+          nodes: query.reports.nodes,
+          totalCount: query.reports.totalCount,
+        };
+      }
+      if (query?.reports.__typename == 'QueryReportsError') {
+        let errorMessage;
+        switch (query.reports.error.__typename) {
+          case 'FailedTranslation':
+            errorMessage = t('report.error-translating', {
+              key: query.reports.error.description,
+            });
+            break;
+          // TODO add never exhaustive error handling if adding more error types to QueryReportError
+          // default:
+          //   noOtherVariants(query.reports.error.__typename);
```

Comment:
Thought would be helpful to leave this here for if and when we add errors? Wasn't sure on this one